### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by Wahaj-Ali**
### Highlights
**Hello Reviewer** 👋 
This PR contains
- unit tests using Rspec.
- TDD approach.

### CODE REVIEW

File: spec/tdd_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  solver = Solver.new
+  number = 5
+  three = 3
+  word = 'open'
+
+  context 'Test the solver class factoiral method.' do
+    it 'it should return the factorial of a number passed as an argument' do
+      factorial = solver.factorial(number)
+      expect(factorial).to eql 120
+    end
```
Comment (tresorsawasawa) [inline spec/tdd_spec.rb:13]: - [x] As required for the project, kindly ensure that The` #factorial` method has tests that handle the following cases:
  * Negative value which raises an exception as an expectation
  * Special case - when input is **0**, the result should equal **1**

File: spec/tdd_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    it 'it should return a string "buzz" when a divisible of 5 is passed as an argument' do
+      string = solver.fizzbuzz(5)
+      expect(string).to eql 'buzz'
+    end
+
+    it 'it should return a the same number as a string when anything other than a divisible of 3 or 5 is passed as an argument' do
+      string = solver.fizzbuzz(7)
+      expect(string).to eql '7'
+    end
+  end
```
Comment (tresorsawasawa) [inline spec/tdd_spec.rb]: - [x]  As required for the project, kindly ensure that The `#fizzbuzz` method has tests that also handle the following case:
  * `Number` is divisible by `3` and `5` with `fizzbuzz` as expectation

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Wahaj-Ali and @rbhatt1999 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/tdd_spec.rb (hunk 1 above)
Comment (Wahaj-Ali) [reply spec/tdd_spec.rb:13]: @tresorsawasawa thank you for letting me know about the short coming. I will make the required change.

File: spec/tdd_spec.rb (hunk 2 above)
Comment (Wahaj-Ali) [reply spec/tdd_spec.rb]: @tresorsawasawa thank you again for letting me know about the test case that I missed. I will surely add it.

File: spec/tdd_spec.rb (hunk 1 above)
Comment (Wahaj-Ali) [reply spec/tdd_spec.rb:13]: Here is the [change](https://github.com/Wahaj-Ali/TDD/pull/1/commits/fd92f5802d1ba339da933ef451f15ad459112605) I made.

File: spec/tdd_spec.rb (hunk 2 above)
Comment (Wahaj-Ali) [reply spec/tdd_spec.rb]: Updated [solution](https://github.com/Wahaj-Ali/TDD/pull/1/commits/944cb54f4bde1de41de55594af3caeace39149d4).

**Review by Munsa1 — APPROVED** [review]
Hi @Wahaj-Ali @rbhatt1999, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made required changes :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Wahaj-Ali** [comment]
Thank you @Munsa1 for approving the PR.

**Pull request merged**